API/common/utils.py

```diff
@@ -265,47 +276,26 @@ def get_section_sizes_from_map(mapfile):
         'total': 0
     }
 
-    if not exists(mapfile):
-        return sizes
-
-    archives = [
-        'libhttpparser.a',
-        'libiotjs.a',
-        'libjerry-core.a',
-        'libjerry-ext.a',
-        'libjerry-port-default.a',
-        'libjerry-port-default-minimal.a',
-        'libtuv.a'
-    ]
+    mapfile = join(builddir, 'linker.map')
+    libdir = join(builddir, 'libs')
 
-    # FIXME: use `ar` command to get the object files from
-    # the static libraries. With the object name list, the
-    # binary size estimation can be more punctual and don't
-    # need this exclude section.
-    exclude_archives = [
-        'libjerry-libm.a',
-        'libjerry-libc.a'
-    ]
+    if not (exists(mapfile) and exists(libdir)):
+        return sizes
 
     data = lumpy.load_map_data(mapfile)
-
     sections = lumpy.parse_to_sections(data)
     # Extract .rodata section from the .text section.
     lumpy.hoist_section(sections, ".text", ".rodata")
 
+    objlist = read_object_files(libdir, _liblist)
+
     for s in sections:
         for section_key in sizes:
             if s['name'] != '.%s' % section_key:
                 continue
 
             for ss in s['contents']:
-                if len(filter(lambda ar: ar in ss['path'], exclude_archives)):
-                    continue
-
-                cobj_in_path = ss['path'].endswith('.c.obj)')
-                arch_in_path = len(filter(lambda ar: '/%s(' % ar in ss['path'], archives))
-
-                if cobj_in_path or arch_in_path:
+                if any(obj in ss['path'] for obj in objlist):
```

Comment:
What about to use a more descriptive variable name instead of `ss`? :)

Reply:
I've modified the algorithm to be more readable.